### PR TITLE
New overload: implementing IModelCacheKeyFactory

### PR DESCRIPTION
**Pull request by joelmandell**
EF Core 6, new overload example for "Design-time DbContext creation"-support, when implementing IModelCacheKeyFactory.

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/EntityFramework.Docs?pullRequest=3305) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/EntityFramework.Docs?pullRequest=3305) before we can accept your contribution.<br/><br/>:x: joelmandell [sign now](https://cla.dotnetfoundation.org/dotnet/EntityFramework.Docs?pullRequest=3305)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/EntityFramework.Docs?pullRequest=3305) it.</sub>

**Comment by joelmandell** [comment]
Regarding this issue: https://github.com/dotnet/efcore/issues/25154

File: entity-framework/core/modeling/dynamic-model.md [hunk 1]
```diff
@@ -21,6 +21,10 @@ The following implementation takes the `UseIntProperty` into account when produc
 
 [!code-csharp[Main](../../../samples/core/Modeling/DynamicModel/DynamicModelCacheKeyFactory.cs?name=DynamicModel)]
 
+If you have to support [Design-time DbContext creation](xref:core/cli/dbcontext-creation)(Needed by, for example, the Migration commands), you have to use the new overload for the Create method. As in the following example:
```
Comment (AndriySvyryd) [inline entity-framework/core/modeling/dynamic-model.md:24]:
```suggestion
Starting with EF Core 6 you have to implement the overload of the Create method that also handles design-time model caching. As in the following example:
```

**Review by AndriySvyryd — APPROVED** [review]

File: samples/core/Modeling/DynamicModel/DynamicModelCacheKeyFactoryDesignTimeSupport.cs [hunk 2]
```diff
@@ -0,0 +1,18 @@
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Infrastructure;
+
+namespace EFModeling.DynamicModel
+{
+    #region DynamicModelDesignTimeSupport
+    public class DynamicModelCacheKeyFactory : IModelCacheKeyFactory
```
Comment (AndriySvyryd) [inline samples/core/Modeling/DynamicModel/DynamicModelCacheKeyFactoryDesignTimeSupport.cs]: Rename this and change `DynamicModel.csproj` to target EF Core 6

**Comment by AndriySvyryd** [comment]
Thanks for your contribution!

**Comment by joelmandell** [comment]
The pleasure was all mine sir 😊

**Pull request merged**